include/mbgl/util/geometry.hpp

```diff
@@ -12,6 +12,16 @@ enum class FeatureType : uint8_t {
     Polygon = 3
 };
 
+struct ToFeatureType {
+    FeatureType operator()(mapbox::geometry::point<int16_t>) const { return FeatureType::Point; }
```

Comment:
These should all take their argument as `const __ &` to avoid copying, and use the CamelCase aliases defined below.